src/libraries/Microsoft.Extensions.Logging.EventSource/src/EventSourceLogger.cs

```diff
@@ -57,23 +58,58 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             {
                 return;
             }
+
+            bool formattedMessageEventEnabled = _eventSource.IsEnabled(EventLevel.Critical, LoggingEventSource.Keywords.FormattedMessage);
+            bool messageEventEnabled = _eventSource.IsEnabled(EventLevel.Critical, LoggingEventSource.Keywords.Message);
+            bool jsonMessageEventEnabled = _eventSource.IsEnabled(EventLevel.Critical, LoggingEventSource.Keywords.JsonMessage);
+
+            if (!formattedMessageEventEnabled
+                && !messageEventEnabled
+                && !jsonMessageEventEnabled)
+            {
+                return;
+            }
+
             string? message = null;
 
+            Activity? activity = Activity.Current;
+            string? activityTraceId;
+            string? activitySpanId;
+            string? activityTraceFlags;
+            if (activity != null && activity.IdFormat == ActivityIdFormat.W3C)
+            {
+                activityTraceId = activity.TraceId.ToHexString();
+                activitySpanId = activity.SpanId.ToHexString();
+                activityTraceFlags = activity.ActivityTraceFlags == ActivityTraceFlags.None
+                    ? "0"
+                    : "1";
+            }
+            else
+            {
+                activityTraceId = null;
```

Comment:
I added another comment that we can intialize these with empty string and avoid checking again in the event emission.

Reply:
Updated